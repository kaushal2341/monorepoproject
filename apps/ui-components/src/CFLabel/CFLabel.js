import React, {PureComponent} from 'react';
import Form from 'react-bootstrap/Form';
import PropTypes from "prop-types";

class CFLabel extends PureComponent {

    addGridPropsIfColumnSetToTrue = column => {
        const {sm, md, lg, xl} = this.props;
        let colProps = {};
        if (column) {
            colProps = sm ? {...colProps, 'sm': sm} : colProps;
            colProps = md ? {...colProps, 'md': md} : colProps;
            colProps = lg ? {...colProps, 'lg': lg} : colProps;
            colProps = xl ? {...colProps, 'xl': xl} : colProps;
        }
        return colProps;
    };

    getPropsForLabel = () => {
        const {_ref, as, bsPrefix,className, column, htmlFor, id, srOnly} = this.props;
        return (
            {
                '_ref': _ref,
                'as': as,
                'bsPrefix': bsPrefix,
                'className':className,
                'column': column,
                'htmlFor': htmlFor,
                'id': "label_".concat(id),
                'srOnly': srOnly,
                ...this.addGridPropsIfColumnSetToTrue(column)
            }
        );
    };

    render() {
        const {labelName} = this.props;
        return (
            <Form.Label
                {...this.getPropsForLabel()}
            >
                {labelName}
            </Form.Label>
        )
    }
}


CFLabel.propTypes = {
    _ref: PropTypes.oneOfType([
        PropTypes.func, // for legacy refs
        PropTypes.shape({current: PropTypes.instanceOf(Element)})
    ]),
    as: PropTypes.elementType,
    bsPrefix: PropTypes.string,
    className: PropTypes.string,
    column: PropTypes.bool,
    htmlFor: PropTypes.string,
    id: PropTypes.string.isRequired,
    labelName: PropTypes.string.isRequired,
    srOnly: PropTypes.bool,
    sm: PropTypes.number,
    md: PropTypes.number,
    lg: PropTypes.number,
    xl: PropTypes.number
};

export default CFLabel;