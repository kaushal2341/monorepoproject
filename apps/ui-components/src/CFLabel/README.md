# CLabel

Form Label made using react-bootstrap's `Form.Label` component.
To use it, first

```jsx harmony
import {CFLabel} from '@cogent/ui-components';
```
Then simply use it,
```jsx harmony
 <CFLabel
     id="name"
     labelName="Name"
     column={true}
     srOnly={false}
     md={4}
 />

```

## CLabel props:
```text
_ ref: ReactRef - The FormGroup ref will be forwarded to the underlying element. Unless the FormGroup is rendered as a composite component, it will be a DOM node, when resolved.
as: elementType - You can use a custom element type for this component.
bsPrefix: string - defaultValue: 'form-group' - Change the underlying component CSS base class name and modifier class names prefix. This is an escape hatch for working with heavily customized bootstrap css.
className: string - Css classname for additional styling.
column: boolean - defaultValue: false - Renders the FormLabel as a <Col> component (accepting all the same props), as well as adding additional styling for horizontal forms.
htmlFor: string - Uses controlId from <FormGroup> if not explicitly specified.
id: string - id of the FormGroup component.
labelName: string - Display name of Label
srOnly: boolean - defaultValue: false - Hides the label visually while still allowing it to be read by assistive technologies.
sm,md,lg,xl : number - To provide column size(grid). To be used with column props set to true.

```
We can pass ``sm``,``lg``,``md`` and ``xl`` props only when column prop is passed with boolean value 'true'.
Example:
```jsx harmony
<CFLabel 
    labelName="Email" 
    column={true} 
    sm={2}
/>
```