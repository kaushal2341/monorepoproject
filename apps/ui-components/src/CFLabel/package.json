{
  "name": "@cogent/c-f-label",
  "version": "0.0.0",
  "main": "CFLabel.js",
  "author": "Sabu Shakya",
  "license": "MIT",
  "private": true
}
