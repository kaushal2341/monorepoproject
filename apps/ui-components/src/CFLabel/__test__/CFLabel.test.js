import CFLabel from "../CFLabel";

expect.addSnapshotSerializer(enzymeSerializer);

describe('CFLabel Component Tests', () => {

    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<CFLabel id="test" labelName="test"/>);
    });

    it('should be defined', () => {
        expect(wrapper.find('CFLabel')).toBeDefined();
    });

    it('should have only one Form.Label component', () => {
        expect(wrapper.find('#label_test').length).toBe(1);
    });

    it('should have a name', () => {
        expect(wrapper.find('#label_test').text()).not.toBe("");
    });

    it('should have all props available', () => {
        let propRequired = [
            '_ref',
            'as',
            'bsPrefix',
            'className',
            'column',
            'htmlFor',
            'id',
            'srOnly'
        ];
        let propsOfFormLabel = Object.keys(wrapper.find('#label_test').props());
        propRequired.forEach((propAvail, i) => (
            expect(propAvail).toContain(propsOfFormLabel[i])
        ));
    });

    it('should pass `sm`,`md`,`lg` and `xl` props only when column prop is passed with boolean value true', () => {
        const colProps = [
            {sm: 4},
            {md: 4},
            {lg: 4},
            {xl: 4}];
        wrapper.setProps({
            column: true
        });
        colProps.forEach(colProp => {
            wrapper.setProps({
                ...colProp
            });
            expect(wrapper.find('#label_test').prop(Object.keys(colProp)[0])).toBe(4);
        });
    });

    describe('Snapshot Testing', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })
    });
});