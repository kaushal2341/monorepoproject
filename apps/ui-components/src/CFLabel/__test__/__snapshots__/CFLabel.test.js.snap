// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CFLabel Component Tests Snapshot Testing should match snapshot 1`] = `
<FormLabel
  column={false}
  id="label_test"
  srOnly={false}
>
  test
</FormLabel>
`;
